rusoto/core/src/client.rs

```diff
@@ -35,6 +36,7 @@ impl Client {
         let inner = Arc::new(ClientInner {
             credentials_provider: Some(Arc::new(credentials_provider)),
             dispatcher: Arc::new(dispatcher),
+            content_encoding: Default::default(),
```

Comment:
what if we move the `content_encoding` to the `Client` and pass it to `Inner` methods only when needed.

Reply:
I checked it out and I need to call `encode()` in `SignAndDispatch` trait and `ClientInner` implements it, not `Client`.

Reply:
So than may be add new constructor, or may be a builder. This way we don't need to mutate it.

Reply:
I have implemented in that way, it is better because we don't need to regenerate other services. Take a look at the new version.